crates/store/re_types_core/src/lib.rs

```diff
@@ -187,3 +187,35 @@ pub fn try_serialize_field<C: crate::Component>(
         }
     }
 }
+
+/// Internal serialization helper for code-generated archetypes.
+///
+/// This generates a correctly sized [`SerializedComponentColumn`] for a given indicator, where the
+/// specified `num_rows` value represents the number of rows in the column.
+#[doc(hidden)] // public so we can access it from re_types too
+pub fn indicator_column<A: Archetype>(
```

Comment:
This can be greatly simplified, and much cheaper as well -> indicators are straight null arrays these days, no lists involved.

Reply:
That does mean making `SerializedComponentColumn` more complicated just for the sake of indicators, i.e. legacy...

Annoying, but there isn't really a choice here.

Reply:
hmm yeah we'd have the same problem in C++ as well (in Python all the indicator handling happens on the Rust side anyways iirc)

Is it really worth it?

Reply:
(re Cpp: `ComponentColumn` there has a `from_indicators` which does pretty much what we're doing here)

Reply:
It's more complicated because `SerializedComponentColumn` would have to carry an erased Array instead, and have two paths for everything. I don't like it.

I'm gonna investigate an alternative: have `send_columns_v2` automagically craft indicator columns as required on the fly. I don't like it either but objectively it is far less invasive.

Reply:
Oh god, this actually is an already existing problem, indicators are wrapped in a `ListArray` when put into chunks (probably to avoid adding a ton of complexity for a dead feature) :facepalm:.

That means:
* Indicators take 4 bytes per row as of today (the outer list-array's offsets).
* What I did is actually the right thing to do, and it makes me very sad.

Man, indicators really gotta go...

Reply:
Goes without saying, this will have to happen a yet another follow up.

Reply:
https://github.com/rerun-io/rerun/issues/8768